python/cuml/metrics/pairwise_distances.pyx

```diff
@@ -0,0 +1,222 @@
+        EucExpandedL2Sqrt "ML::Distance::DistanceType::EucExpandedL2Sqrt"
+        EucExpandedCosine "ML::Distance::DistanceType::EucExpandedCosine"
+        EucUnexpandedL1 "ML::Distance::DistanceType::EucUnexpandedL1"
+        EucUnexpandedL2 "ML::Distance::DistanceType::EucUnexpandedL2"
+        EucUnexpandedL2Sqrt "ML::Distance::DistanceType::EucUnexpandedL2Sqrt"
+
+cdef extern from "cuml/metrics/metrics.hpp" namespace "ML::Metrics":
+    void pairwiseDistance(const cumlHandle &handle, const double *x,
+                          const double *y, double *dist, int m, int n, int k,
+                          DistanceType metric, bool isRowMajor) except +
+    void pairwiseDistance(const cumlHandle &handle, const float *x,
+                          const float *y, float *dist, int m, int n, int k,
+                          DistanceType metric, bool isRowMajor) except +
+
+
+def determine_metric(metric_str):
+
+    # Available options in scikit-learn and their pairs. See
+    # sklearn.metrics.pairwise.PAIRWISE_DISTANCE_FUNCTIONS:
+    # 'cityblock': EucUnexpandedL1
+    # 'cosine': EucExpandedCosine
+    # 'euclidean': EucUnexpandedL2Sqrt
+    # 'haversine': N/A
+    # 'l2': EucUnexpandedL2Sqrt
+    # 'l1': EucUnexpandedL1
+    # 'manhattan': EucUnexpandedL1
+    # 'nan_euclidean': N/A
+    # Note: many are duplicates following this:
+    # https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/metrics/pairwise.py#L1321
+
+    if metric_str == 'cityblock':
+        return DistanceType.EucUnexpandedL1
+    elif metric_str == 'cosine':
+        return DistanceType.EucExpandedCosine
+    elif metric_str == 'euclidean':
+        return DistanceType.EucUnexpandedL2Sqrt
+    elif metric_str == 'haversine':
+        raise ValueError(" The metric: '{}', is not supported at this time."
+                         .format(metric_str))
+    elif metric_str == 'l2':
+        return DistanceType.EucUnexpandedL2Sqrt
+    elif metric_str == 'l1':
+        return DistanceType.EucUnexpandedL1
+    elif metric_str == 'manhattan':
+        return DistanceType.EucUnexpandedL1
+    elif metric_str == 'nan_euclidean':
+        raise ValueError(" The metric: '{}', is not supported at this time."
+                         .format(metric_str))
+    else:
+        raise ValueError("Unknown metric: {}".format(metric_str))
+
+
+@with_cupy_rmm
+def pairwise_distances(X, Y=None, metric="euclidean", handle=None,
+                       convert_dtype=True, **kwds):
+    """
+    Compute the distance matrix from a vector array X and optional Y.
+
+    This method takes either one or two vector arrays, and returns
+    a distance matrix.
+
+    If Y is given (default is None), then the returned matrix is the pairwise
+    distance between the arrays from both X and Y.
+
+    Valid values for metric are:
+
+    - From scikit-learn: ['cityblock', 'cosine', 'euclidean', 'haversine',
+                          'l1', 'l2', 'manhattan'].
```

Comment:
Just to get another quick win, what do you think about adding `sqeuclidean` to this list? It's just the eudlidean without the sqrt at the end.

Reply:
I think it's a good idea. Should be trivial to map `sqeuclidean` to `DistanceType.EucUnexpandedL2` and add a few tests.

Reply:
This was as easy as we expected. Added and all tests pass.